docs/example_go.md

```diff
@@ -3,29 +3,273 @@ title: "Example: Go"
+trigger local jobs, or run local servers. We add a `local_resource` to our
+[Tiltfile](https://github.com/windmilleng/tilt-example-go/blob/master/1-measured/Tiltfile)
+that records the start time in a Go file. We've also modified our server itself to
+read that start time and print the time elapsed.
+
+```python
+k8s_resource(
+    'example-go', 
+    port_forwards=8000, 
+    resource_deps=['deploy'])
+
+# Records the current time, then kicks off a server update.
+# Normally, you would let Tilt do deploys automatically, but this
+# shows you how to set up a custom workflow that measures it.
+local_resource(
+    'deploy',
+    './record-start-time.sh',
+)
+```
+
+The `local_resource()` call creates a local resource named `deploy`. The second
+argument is the script that it runs.
+
+Let's click the button on the `deploy` resource and see what happens!
+
+<figure>
+  <a class="is-image" href="https://cloud.tilt.dev/snapshot/AbakyOYLWWvW6J2BJFo=">
+    <img src="assets/docimg/example-static-html-image-2.png">
+  </a>
+  <figcaption>Step 1 complete. Click the screenshot to see an interactive snapshot.</figcaption>
+</figure>
+
+| Approach | Deploy Time |
+|---|---|
+| Naive | 4.2s |
+
+Can we do better?
+
+## Step 2: Let's Optimize for the Go Compiler
+
+When we make a change to a file, we currently copy all the source files to a
+container, download the dependencies, and re-compile the Go binary from scratch.
+
+But the Go compiler team has done a lot of work to make incremental compiles
+fast. How can we better use the Go tools how they're meant to be used?
+
+With `local_resource`, we can compile the Go binary locally, and copy the binary to
+the container.
+
+Here's our [new Tiltfile](https://github.com/windmilleng/tilt-example-go/blob/master/2-optimized/Tiltfile) 
+with the following new code:
+
+```python
+local_resource(
+  'example-go-compile',
+  'CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o build/tilt-example-go ./',
+  deps=['./main.go', './pkg'])
+
+docker_build(
+  'example-go-image',
+  '.',
+  dockerfile='deployments/Dockerfile',
+  only=[
+    './build',
+    './web',
+  ])
+```
+
+We've added a `local_resource()` that compiles the Go binary locally for our
+Linux container.
+
+We've added a new `only` parameter to `docker_build()`. This makes sure that the
+only directory in the Docker build context are the `./build` directory (for the
+compiled binary) and the `./web` directory (for the template and image files).
+
+We've also modified the Dockerfile to only copy the `./build` directory.
+
+Let's see what this looks like!
+
+<figure>
+  <a class="is-image" href="https://cloud.tilt.dev/snapshot/AcK4yOYLQObve-87cQ4=">
+    <img src="assets/docimg/example-go-image-3.png">
+  </a>
+  <figcaption>Step 2 complete. Click the screenshot to see an interactive snapshot.</figcaption>
+</figure>
+
+| Approach | Deploy Time |
+|---|---|
+| Naive | 4.2s |
+| Local Compile | 3.5s |
+
+
+## Step 3: Let's Optimize It
```

Comment:
i'm curious why you did local compile first, THEN live update, can you explain? (i don't think it's bad, it's just different than how we've organized things in the past)

Reply:
good question! mainly following the outline. But I can think of a few reasons to do it this way:

- Adding `live_update` is a good capstone for a Tilt tutorial

- A lot of people in the Go community have started using this project for Go builds: 

https://github.com/thockin/go-build-template

which uses this Dockerfile:

https://github.com/thockin/go-build-template/blob/master/Dockerfile.in

which is consistent with the approach in this tutorial, so there's some prior art